clash-prelude/src/Clash/Class/BitPack.hs

```diff
@@ -134,6 +135,19 @@ packXWith f x =
                                 (\(XException _) -> return undefined#))
 {-# NOINLINE packXWith #-}
 
+-- | Check if the bit patterns of two values are similar. For most types this
+-- is equivalent to '(==)', but for types which contain 'BitVector' or 'Bit'
+-- it can behave more leniantly when certain bits are _don't care_. See
+-- 'Clash.Sized.Internal.BitVector.isLike#' for more information.
+--
+isLike
+  :: (BitPack a)
+  => a
+  -> a
```

Comment:
Have you thought about writing:

```haskell
isLike :: (BitPack a, BitPack b, BitSize a ~ BitSize b) => a -> b -> Bool
```

I'm on the fence on whether this is a good idea.